### PR TITLE
PXP-634:  [CLI] Add welcome before showing TUI panels 

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary

Develop a welcome screen in the TUI to check the following pre-conditions. If all conditions are passed, we can start showing data in the TUI, if not, show the instructions telling user what to do.
Screenshot 2023-11-21 at 10 43 34 AM

Ticket: [https://mindvalley.atlassian.net/browse/PXP-634](https://mindvalley.atlassian.net/browse/PXP-634url)

## What's Changed

Pre-conditions:
- [x] Add new endpoint for google token verification
- [x] Load welcome screen on start instead of main screen
- [x] Check if the user has a valid Okta token.
- [x] Check the user has a valid Google token.

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/173#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: sdk/src/services/vault/client.rs
Comment (mfauzaan) [inline sdk/src/services/vault/client.rs:60]: New rust version enforce this

Ref issue: https://github.com/rust-lang/rust/issues/115010

File: cli/src/output/tokenizer.rs
Comment (mfauzaan) [inline cli/src/output/tokenizer.rs:191]: Clippy fix

ref: https://rust-lang.github.io/rust-clippy/master/#/clippy::collapsible-el

File: cli/src/commands/tui/ui/welcome.rs
Comment (onimsha) [inline cli/src/commands/tui/ui/welcome.rs:123]: I think this message is a little bit misleading. It will always ask user to re-init even if they are already done the initialisation step. You can see it in the screenshot below. 

image

**Comment by onimsha** [comment]
We're not handling the case of Google Logging well in the Splash screen. Below is the screenshot 

image

If we are to show instruction here, it should be formatted properly

**Comment by onimsha** [comment]
Also based on my observation, it looks like we're doing something with the networking logic, from API Logs, there're many requested being stacked at the same time. This behaviour does not happen in the `main` branch.  

image

**Comment by onimsha** [comment]
This is a dead end. We should tell user what to do next. Unlike Okta auth, running `wukong login` is not going to send user to Google Auth. 

image

Can we display a text box to tell user to press `Enter` to redirect to Google Auth ?

**Comment by sauron-droid** [comment]
PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: cli/src/commands/google/login.rs
Comment (jk-gan) [inline cli/src/commands/google/login.rs:4]: I believe this is typo, should be `loader` right?

File: cli/src/commands/tui/app.rs
Comment (jk-gan) [inline cli/src/commands/tui/app.rs:105]: Do they need to be `Option`?

**Review by jk-gan — CHANGES_REQUESTED** [review]

File: cli/src/commands/tui/app.rs
Comment (mfauzaan) [reply cli/src/commands/tui/app.rs:105]: Yes, reason being is that it has 3 states, not verified, verifying and failed
So i can see this can be accomplished by having an Option or 3 enums. 

^ above states is used to display the UI for the user

File: cli/src/auth/google_cloud.rs
Comment (jk-gan) [inline cli/src/auth/google_cloud.rs:17]: is this config "file" or config "path"?

**Review by jk-gan — CHANGES_REQUESTED** [review]

File: cli/src/auth/google_cloud.rs
Comment (mfauzaan) [reply cli/src/auth/google_cloud.rs:17]: hmm, i think it should be config_path

**Review by jk-gan — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/173#pullrequestreview-1754475737" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/173#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — COMMENTED** [review]
LGTM ❤️

**Pull request merged**
